urllib3/connectionpool.py

```diff
@@ -98,6 +98,23 @@ def close(self):
 # This is taken from http://hg.python.org/cpython/file/7aaba721ebc0/Lib/socket.py#l252
 _blocking_errnos = set([errno.EAGAIN, errno.EWOULDBLOCK])
 
+# CPython 2.7 supports an opt-in parameter for response buffering
+# in HTTPConnection.getresponse
+# Rather than checking on each request, we inspect the code object
+# (if available) at import and only pass the argument if it appears
+# in the list of variable names
+def _request_httplib_response_buffering():
```

Comment:
So @ncoghlan it should be noted that if we want to continue with this current mechanism (which is fine by me), we should absolutely reproduce your logic above in a code comment. This is going to be tricky code to maintain, particularly for anyone less versed in Python internals than you (which is to say 98% of Python programmers of any stripe), so understanding what is being done here and the various ways in which this can return the correct result are really important.